kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/NetworkConfiguration.java

```diff
@@ -186,10 +187,21 @@ public void addNetConfig(String interfaceName, NetInterfaceType netInterfaceType
 
         s_logger.trace("Adding a netConfig: {}", netConfig);
         for (NetInterfaceAddressConfig netInterfaceAddressConfig : netInterfaceAddressConfigs) {
-            NetInterfaceAddressConfigImpl netInterfaceAddressConfigImpl = (NetInterfaceAddressConfigImpl) netInterfaceAddressConfig;
             List<NetConfig> netConfigs = netInterfaceAddressConfig.getConfigs();
             netConfigs.add(netConfig);
-            netInterfaceAddressConfigImpl.setNetConfigs(netConfigs);
+
```

Comment:
I just fixed the code so that it no longer throws an exception with the other implementations of that interface. But your suggestion is even better.

With this now fixed I was able to remove some tests and supporting classes, because they were no longer needed.